### PR TITLE
Add package groff

### PR DESCRIPTION
**Pull request by Maxython**
#2056

### CODE REVIEW

**Comment by Maxython** [comment]
The libxaw package is needed to compile, but it is in termux-x11.

**Comment by Maxython** [comment]
Porting to termux-x11

**Comment by Maxython** [comment]
https://github.com/termux/x11-packages/pull/443

**Comment by xeffyr** [comment]
From what time `groff` became the X11 package?

**Comment by xeffyr** [comment]
Don't look at https://archlinux.org/packages/core/x86_64/groff/. Termux goes its own way, neither Debian-specific nor Arch, for all package types.

**Comment by Maxython** [comment]
> From what time `groff` became the X11 package?

Yes, it looks strange. I did this to make `groff` compile with `libxaw`, it just doesn't see this library from this repo.

File: packages/groff/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+TERMUX_PKG_HOMEPAGE=https://www.gnu.org/software/groff/
+TERMUX_PKG_DESCRIPTION="typesetting system that reads plain text mixed with formatting commands and produces formatted output"
+TERMUX_PKG_LICENSE="GPL-3.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=1.22.4
+TERMUX_PKG_SRCURL=https://ftp.gnu.org/gnu/groff/groff-${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=e78e7b4cb7dec310849004fa88847c44701e8d133b5d4c13057d876c1bad0293
+TERMUX_PKG_DEPENDS="perl"
+TERMUX_PKG_BUILD_DEPENDS="netpbm, psutils, libice, libsm, libxau, libxdmcp, libxcb"
```
Comment (xeffyr) [inline packages/groff/build.sh:9]: Remove netpbm, psutils and x11 packages from there. `gxditview` is not a critical functionality also switching to `man-db` from `mandoc` is not planned currently.

**Review by xeffyr — CHANGES_REQUESTED** [review]

**Comment by xeffyr** [comment]
Try `TERMUX_PKG_BUILD_IN_SRC=true`

**Comment by Maxython** [comment]
Thank

**Comment by Maxython** [comment]
Well, despite the fact that it compiled, it did not pass the test.  I also made a compilation for myself, the result: some commands do not work, but they are united by one conclusion.  
```
FORTIFY: %n not allowed on Android                           
Aborted
```
Now guess what it is, it's either the code or the compiler.

**Comment by xeffyr** [comment]
@Maxython Look existing patches. That's due to old gnulib which doesn't have fix for Android 10.

Sample patch made for `m4`, now removed after upstream update:
```.diff
diff --git a/packages/m4/lib-vasnprintf.c.patch b/packages/m4/lib-vasnprintf.c.patch
new file mode 100644
index 000000000..095184b5b
--- /dev/null
+++ b/packages/m4/lib-vasnprintf.c.patch
@@ -0,0 +1,12 @@
+diff -u -r ../m4-1.4.18/lib/vasnprintf.c ./lib/vasnprintf.c
+--- ../m4-1.4.18/lib/vasnprintf.c	2016-12-31 13:54:42.000000000 +0000
++++ ./lib/vasnprintf.c	2019-10-02 19:32:13.576211922 +0000
+@@ -4858,7 +4858,7 @@
+ #endif
+                   *fbp = dp->conversion;
+ #if USE_SNPRINTF
+-# if !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__))
++# if !(((__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 3)) && !defined __UCLIBC__) || ((defined _WIN32 || defined __WIN32__) && ! defined __CYGWIN__) || defined(__ANDROID__))
+                 fbp[1] = '%';
+                 fbp[2] = 'n';
+                 fbp[3] = '\0';
```

**Comment by xeffyr** [comment]
You are attempting to run cross-compiled binary. This is not possible as our build environment is strictly Ubuntu x86_64.
```
  GROFF    doc/meintro.ps
/bin/bash: line 1: /home/builder/.termux-build/groff/src/groff: cannot execute binary file: Exec format error
```

You need to perform a host build (relevant examples are available in ./packages/*) and add path to host-compatible `groff` to PATH. You may need to patch build system to avoid picking cross-compiled groff.

**Comment by Maxython** [comment]
I tried installing and noticed the first problem, a conflict with the `man` package because of the `soelim` command.
Screenshot_20211001-212545_Termux

**Comment by Maxython** [comment]
> You need to perform a host build (relevant examples are available in ./packages/*) and add path to host-compatible groff to PATH.

I just turned off these tests, but your version sounds more correct.

**Comment by Maxython** [comment]
@xeffyr, such a situation with `groff` and c `man` is so to itself.  I've already set it up so that `groff` doesn't conflict with `man`.

**Comment by Maxython** [comment]
https://github.com/termux/termux-packages/pull/7630

**Pull request merged**